app/os_macos.go

```diff
@@ -557,6 +562,22 @@ func gio_onMouse(view, evt C.CFTypeRef, cdir C.int, cbtn C.NSInteger, x, y, dx,
 	})
 }
 
+//export gio_onExternalDrop
+func gio_onExternalDrop(view C.CFTypeRef, path *C.char) {
+	fileUrl := C.GoString(path)
+	w := mustView(view)
+
+	fileExtension := filepath.Ext(fileUrl)
+	mime := mime.TypeByExtension(fileExtension)
```

Comment:
I think it's a bit weird to do detection by file extension here. The detection might not always work (lets say for example, the user is dropping an image file with an unknown extension, or no extension), and in order to receive 'any file' the app would need to register every possible mime type.

Reply:
Correct, but let's support "any file" then, in the form of mime type patterns (e.g. "*" or "image/*"). Not for this PR, of course.